### PR TITLE
Spec migration

### PR DESCRIPTION
**Pull request by adlerjohn**
Ref: #650

Staging branch for specs migration. Draft PR so it can be tracked.

### CODE REVIEW

**Comment by rootulp** [comment]
hmm I can't merge `main` into this branch

Screen Shot 2022-09-28 at 2 55 15 PM


I wanted to merge `main` to pick up the new CODEOWNERS file

**Comment by musalbas** [comment]
What's the status of this?

**Comment by rootulp** [comment]
@musalbas apologies for delayed response. @adlerjohn and I triaged issues in https://github.com/celestiaorg/celestia-app/issues/650 today. 

I think that the `specs-staging` branch is already a more accurate representation of the specs than https://celestiaorg.github.io/celestia-specs/latest/index.html so I'd like to merge `specs-staging` to `main` ASAP. But if we want to block this merge on updating specs to match the existing implementation, we can block on all the issues in the "Pure specs" section.

**Comment by musalbas** [comment]
What's the latest with this and generally adding specs to celestia-app/core now?

**Comment by rootulp** [comment]
This branch is the most up to date version of the specs but it's not publicly visible yet. IMO we should merge this branch to `main`, make it publicly visible, and take down the existing specs site at https://celestiaorg.github.io/celestia-specs/latest/index.html

The remaining specs issues are tracked under https://github.com/celestiaorg/celestia-app/issues/650 . We intend on prioritizing specs and documentation work after the core/app feature freeze for incentivized testnet.

**Comment by cmwaters** [comment]
I have some general comments that can possible be followed up at a later point:

- Can we break up data structures into smaller more self-explanatory files so that parts can be easily accessed in the table of contents. I also think it's less intuitive to explain a spec by just dumping all the data structures that are included in it. Would prefer more of a hierarchical function-orientated tree.
- Can we add a section explaining how the "celestia app" fits in with the rest of the system and include links so people can be easily redirected if they're interested with something in "celestia node" for example. Furthermore, I think calling it "celestia app" can be better worded. As this isn't the specification of the entire system, rather a component of it, we should look to distinguish it functionally. Call it "celestia data publication specification" (to complement "celestia data availability specfication") if we want to be deliberate about having two networks. 
- I don't think `Rationale` should be a separate section but should be embedded right next to the part in the spec it refers to. I should read what a certain protocol is and directly after why it is so. 
- It's evident that the specification includes not just novel work specific to Celestia but the components that it is built on top of. There is a lot of work on Tendermint's consensus algorithm (such as block validity, leader selection) and how the validator set is controlled. What about other components like governance, IBC, vesting, upgrades that still seem to be missing. It seems that not all state transition messages have been documented.
- Is QGB spec somewhere else? 
- Can we have some preamble/context in the top level pages to each of the subpages.
- Are we sure we want a copy of many of the proto files in the spec as well? These are not the canonical ones that get used in generation and will constantly become outdated.

**Review by evan-forbes — COMMENTED** [review]
are the markdown linter complaints valid?

```
  ERROR: 2 dead links found!
  [✖] .consensus.md#constants → Status: 400
  [✖] ../rationale/distributing_rewards.md → Status: 400
 ```
 
 other than that I think we're okay to merge (and actually merge, not squash!)
 
 we should also add @cmwaters feedback into the spec epic as @rootulp mentioned because I think its really good

**Comment by rootulp** [comment]
Will merge this by EOD tomorrow if no more feedback.

File: .github/workflows/gh-pages.yml
Comment (MSevey) [inline .github/workflows/gh-pages.yml:24]: [nit] I'm pretty sure you just need to check the `github.event_name` since we are already filtering to only triggering on the `main` branch for `push` events up top. But if you'd prefer to keep this here as an additional level of protection that is fine. 

```suggestion
        if: github.event_name == 'push'
```

**Comment by evan-forbes** [comment]
its weird that the docker builds are still failing here despite passing in other PRs, we didn't change anything related here afaict

File: .github/workflows/gh-pages.yml
Comment (rootulp) [reply .github/workflows/gh-pages.yml:24]: good point! Will adopt your suggestion

File: .github/workflows/gh-pages.yml
Comment (rootulp) [reply .github/workflows/gh-pages.yml:1]: good idea! Created https://github.com/celestiaorg/celestia-app/issues/1495

**Review by MSevey — APPROVED** [review]

**Pull request merged**
